lib/include/archive_private.h

```diff
@@ -0,0 +1,120 @@
+#ifndef _LIBDISIR_PRIVATE_ARCHIVE_H
+#define _LIBDISIR_PRIVATE_ARCHIVE_H
+
+#include "tinytoml/toml.h"
+#include <disir/archive.h>
+#include <set>
+#include <libgen.h>
+
+//! Convenience macros:
+//! toml key for a backend table
+#define ATTRIBUTE_KEY_BACKEND "backend"
+//! toml key for entry_id
+#define ATTRIBUTE_KEY_ID "id"
+//! toml key for gourp table
+#define ATTRIBUTE_KEY_GROUPS "groups"
+//! name of metadata file in archive
+#define METADATA_FILENAME "metadata.toml"
+#define TEMP_ARCHIVE_PATH "/tmp/archive.disir.tmp"
```

Comment:
This cannot be used like this.
We will need to use a thread_local_storage variable instead that is generated for each new archive. (https://stackoverflow.com/a/15698197)
Ideally, we would pass this path throughout the API somehow. The scenario here is that one should be able to programmatically export two archives concurrently. Even two different processes using disir at the same time will
collide on export here.